app/main.js

```diff
@@ -731,16 +735,21 @@ function getTrayMenu () {
       createPreferencesWindow()
     }
   }, {
-    type: 'separator'
+    type: 'separator',
+    visible: !settings.get('hardcoreMode')
```

Comment:
Not sure about hiding Quit button. That's something that should always be there, it's basic functionality.

Reply:
What do you think about adding a new config item then?
Breaking hardcore mode into separate config items such as: disable reset breaks, disable skip & postpone breaks (it doesn't make that sense, because we already have strict modes), disable quit app features, disable pause breaks.

In my opinion, real hardcore mode makes it impossible to avoid breaks.
I really need this feature in the implemented way because I always find ways to avoid breaks. 😁

Implementing separate config items seems a reasonable way to handle this issue because then everyone can decide which features they want to hide/disable.

Reply:
I am pretty sure I don't wanna multiple settings, it would make every new feature more complicated. I've made that mistake few times already...

I think we need to move a step back and spend some time thinking what does "hardcore mode" means...

Reply:
For me Hardcore mode means the following: It doesn't allow you to avoid mini/long breaks in any way. It not just disables skipping breaks, it also hides the reset & pause break features, and making it almost impossible to quit/skip breaks and the app itself.

Reply:
So I was thinking last few days and here's what I am leaning towards now: 
- having two settings for one thing is bad idea, so I wouldn't let this mode change postpone/end breaks functionality. User can go hardcore by disabling both of them right now
- selectively picking what to hide in tray seems wrong to me as well, different user will have different opinions what hardcore means and they will be asking me to change the behavior 

So what about adding simple `tray`, that would hide/show whole tray? In this way there is no discussion what hardcore means, we simply have option to hide whole tray. 

Considering that tray is then gone, user needs to go and edit settings file. Pretty hardcore for me :robot:

Reply:
Hiding the tray icon doesn't seem like a good solution for me because that's the one and only thing that displays how long time if left until a next break.

However if we can meet in the middle line that would be a good compromise:
Show the tray icon but disable the tray menu completely. So the display of remaining time until next mini/long breaks are shown in the tooltip, but you can't change anything from the menu/quit the app.

What do you think of this?

Reply:
Let me think a bit more :D Maybe I think of something but probably that's it.

Reply:
Did you find a better solution for this?

Reply:
Not really :) Let's add `trayMenu` `true/false` that will hide tray menu. But let's keep the status messages as not every platform is bug-free with tray tooltip.

Reply:
What do you mean by status messages?
The one which is in the top of the tray menu?

So, should I just hide the menu items in the tray menu, except the top ones which shows remaining time?

How should we turn on/off hardcore mode via the UI? When I originally developed this feature I also implemented a checkbox element for HC mode in Contributor prefs window.
If the Preferences tray menu item is also hidden, then you won't be able to turn off hardcore mode from the UI. Any ideas how to solve this?

Reply:
Yeah, I mean those messages at the top of the menu. 

Well, I would have it only in the file. Makes sense for me to complicate setting/unsetting it :)